policy/scan/scan.proto

```diff
@@ -0,0 +1,110 @@
+syntax = "proto3";
+
+import "motor/inventory/v1/inventory.proto";
+import "motor/providers/provider.proto";
+import "google/protobuf/struct.proto";
+import "policy/cnspec_policy.proto";
+
+package cnspec.policy.scan;
+option go_package = "go.mondoo.com/cnspec/policy/scan";
+
+service Scan {
+  // Run executes a scan immediately and reports the result to Mondoo Platform
+  rpc Run(Job) returns (ScanResult) {}
+  // RunIncognito executes a scan immediately and does not report the results to Mondoo Platform
+  rpc RunIncognito(Job) returns (ScanResult) {}
+  // Schedule put a scan job onto the queue for later execution
+  rpc Schedule(Job) returns (Empty) {}
+
+  // RunAdmissionReview scan a Kubernetes AdmissionReview object
+  rpc RunAdmissionReview(AdmissionReviewJob) returns (ScanResult) {}
+
+  // GarbageCollectAssets wraps the underlying DeleteAssets() to allow
+  // specifying filters to identify and delete assets en masse
+  rpc GarbageCollectAssets(GarbageCollectOptions) returns (Empty) {}
+}
+
+// Defines the possible outputs for a scan. If NONE is used, then no output is returned. If RAW is used,
+// then a raw object ReportCollection object is returned.
+enum ReportType {
+  FULL = 0;
+  ERROR = 1;
+  NONE = 2;
+}
+
+message Empty {}
+
+message Job {
+  cnquery.motor.inventory.v1.Inventory inventory = 1;
```

Comment:
Running this
```
{
  "reportType": 1,
  "inventory": {
    "spec": {
      "assets": [
         {
           "connections": [
               {
                 "backend": 14,
                 "options": {
                  "tenant-id":"d9abc6fc-fd88-4480-a931-2f7939adbac2",
                  "client-id": "03bb7244-e0a6-4abd-bb63-045267d25304",
                  "client-secret": "~~"
                 }
               }
            ]
         }
      ]
    }
  }
}
``` against the scan API results in no assets discovered, since I do not provide any target. Do we want to make this consistent with the CLI scan in the sense that no explicit target = auto?

Reply:
Discussed with @imilchev, best to leave this explicit as it's an API